plugins/woocommerce/includes/wc-account-functions.php

```diff
@@ -176,11 +176,13 @@ function wc_get_account_endpoint_url( $endpoint ) {
 		return wc_get_page_permalink( 'myaccount' );
 	}
 
+	$url = wc_get_endpoint_url( $endpoint, '', wc_get_page_permalink( 'myaccount' ) );
+
 	if ( 'customer-logout' === $endpoint ) {
-		return wc_logout_url();
+		return wp_nonce_url( $url, 'customer-logout' );
 	}
```

Comment:
⚠️ This change broke the logout link on existing sites that were using `wc_logout_url()` before to generate the link to the logout page.

Instead of getting logged out, the user is presented with a question to confirm they want to log out.  Clicking the link in the message repeats the same page response, but the user is never logged out.  On top, this is all happening on the `wp-login.php` front controller of WordPress Core instead of the WooCommerce endpoint.

This is because the `customer-logout` endpoint is using a different nonce than the `wp-login.php` front controller.  And because `wc_logout_url()` calls straight into `wp_logout_url()` now, the unexpected endpoint is loading.

Suggestions:
1. This change should have been announced with a much higher severity, because it is not backwards compatible and breaks existing sites.
2. The nonce of the `customer-logout` endpoint could at least have used the same action as the `wp-login.php?action=logout` endpoint; i.e., `'log-out'`, so at least the user's intention is executed, even if on the wrong controller.

It was anything but trivial to find the cause, so silence on this PR may not indicate that no one is affected.

Reply:
For others affected by this change, the remedy is to adjust the logout URL as follows in your plugins and themes:

```diff
-    $context['logout_url'] = wc_logout_url();
+    $context['logout_url'] = wc_get_account_endpoint_url('customer-logout');
```

Reply:
I'm not fully up to memory with this PR. @mikejolley would it be possibly to modify `wc_logout_url` so it continues to work?

We might include that in 9.5

Reply:
@senadir I was aware using the function without nonce would trigger a confirmation notice to confirm logout, but not that there was a problem with the nonces. Let me check.

Reply:
If you simulate log-out without nonce by navigating to:

```
yourstore.com/my-account/customer-logout
```

You see the prompt/notice (expected). The link in the notice contains a valid logout url with nonce:

```
yourstore.com/wp-login.php?action=logout&redirect_to=https%3A%2F%2Fstore.local%2Fmy-account%2F&_wpnonce=XXX
```

For `wc_logout_url()`, I can see it outputs a valid link with nonce too. This doesn't go via the `customer-logout` endpoint:

```
yourstore.com/wp-login.php?action=logout&redirect_to=https%3A%2F%2Fstore.local%2Fmy-account%2F&_wpnonce=xxx
```

To clarify, links that  go via `yourstore.com/my-account/customer-logout` without nonce now require confirmation.
**This cannot be circumvented because that would allow anyone to trigger logout without using a nonce at all thus rendering all logout nonces pointless.** The confirmation notice is an unfortunate step, but its not breaking.

I also tested `wc_get_account_endpoint_url( 'customer-logout' )` which is not being used any longer in core. This gave me a nonced URL which worked. The handling for that is in here:

https://github.com/woocommerce/woocommerce/blob/05f70533bc6fdb280d7520ccfcb0059128cb74e0/plugins/woocommerce/includes/wc-template-functions.php#L37-L42

```php
if ( ! empty( $_REQUEST['_wpnonce'] ) && wp_verify_nonce( sanitize_key( $_REQUEST['_wpnonce'] ), 'customer-logout' ) ) {
		wp_logout();
		wp_safe_redirect( wc_get_logout_redirect_url() );
		exit;
}
```

This seems fine to me. No bug here @senadir